src/test/java/core/basesyntax/service/TransactionParserServiceTest.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.impl.TransactionParserServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+public class TransactionParserServiceTest {
+    @Test
+    public void transactionParser_parse_ok() {
+        List<String> read = new ArrayList<>();
+        read.add("action,fruit,amount");
+        read.add("b,banana,10");
+        read.add("b,apple,20");
+        read.add("p,banana,5");
+        read.add("s,apple,5");
+        List<FruitTransaction> expected = new ArrayList<>();
+        expected.add(new FruitTransaction(FruitTransaction.Operation.getOperationByCode("b"),
+                "banana",
+                10));
+        expected.add(new FruitTransaction(FruitTransaction.Operation.getOperationByCode("b"),
+                "apple",
+                20));
+        expected.add(new FruitTransaction(FruitTransaction.Operation.getOperationByCode("p"),
+                "banana",
+                5));
+        expected.add(new FruitTransaction(FruitTransaction.Operation.getOperationByCode("s"),
+                "apple",
+                5));
+        List<FruitTransaction> actual = new TransactionParserServiceImpl().parse(read);
```

Comment:
TransactionParserServiceImpl should be created the same way as I explained previously. Fix it and similar issues in all places

Reply:
not fixed